### ISSUE TITLE
Crash: NullPointerException on App Startup

### ISSUE THREAD

**Issue by wsdt** [DylanVann/react-native-fast-image#536]
**Describe the bug**
App crashes on some devices because of a Nullpointer-Exception. 
Might be related to: https://github.com/DylanVann/react-native-fast-image/issues/165 and https://github.com/DylanVann/react-native-fast-image/issues/523

**Device**
Brand: LGE
Model: Nexus 5X
Orientation: Portrait
RAM free: 3.27 GB
Disk free: 1.98 GB

**Operating System**
Version: 6.0.1
Orientation: Portrait
Rooted: Yes

`Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference
       at com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents + 51(FastImageOkHttpProgressGlideModule.java:51)
       at com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents + 37(GeneratedAppGlideModuleImpl.java:37)
       at com.bumptech.glide.Glide.initializeGlide + 273(Glide.java:273)
       at com.bumptech.glide.Glide.initializeGlide + 223(Glide.java:223)
       at com.bumptech.glide.Glide.checkAndInitializeGlide + 184(Glide.java:184)
       at com.bumptech.glide.Glide.get + 168(Glide.java:168)
       at com.bumptech.glide.Glide.getRetriever + 689(Glide.java:689)
       at com.bumptech.glide.Glide.with + 716(Glide.java:716)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance + 51(FastImageViewManager.java:51)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance + 33(FastImageViewManager.java:33)
       at com.facebook.react.uimanager.ViewManager.createViewInstanceWithProps + 119(ViewManager.java:119)
       at com.facebook.react.uimanager.ViewManager.createViewWithProps + 66(ViewManager.java:66)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.createView + 259(NativeViewHierarchyManager.java:259)
       at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute + 198(UIViewOperationQueue.java:198)
       at com.facebook.react.uimanager.UIViewOperationQueue$1.run + 838(UIViewOperationQueue.java:838)
       at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches + 952(UIViewOperationQueue.java:952)
       at com.facebook.react.uimanager.UIViewOperationQueue.access$2200 + 44(UIViewOperationQueue.java:44)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded + 1012(UIViewOperationQueue.java:1012)
       at com.facebook.react.uimanager.GuardedFrameCallback.doFrame + 29(GuardedFrameCallback.java:29)
       at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame + 172(ReactChoreographer.java:172)
       at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame + 84(ChoreographerCompat.java:84)
       at android.view.Choreographer$CallbackRecord.run + 856(Choreographer.java:856)
       at android.view.Choreographer.doCallbacks + 670(Choreographer.java:670)
       at android.view.Choreographer.doFrame + 603(Choreographer.java:603)
       at android.view.Choreographer$FrameDisplayEventReceiver.run + 844(Choreographer.java:844)
       at android.os.Handler.handleCallback + 739(Handler.java:739)
       at android.os.Handler.dispatchMessage + 95(Handler.java:95)
       at android.os.Looper.loop + 148(Looper.java:148)
       at android.app.ActivityThread.main + 5421(ActivityThread.java:5421)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 726(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main + 616(ZygoteInit.java:616)`

**To Reproduce**
On Startup (not all devices). 

**Expected behavior**
No error

**Screenshots**
App crash, not available

**Dependency versions**

- React Native version: 0.60.4
- React Native Fast Image version: 7.0.2

**Comment by halilb** [comment]
We have the exact same crash happening on Nexus 5x devices. I could not find any information in the Glide repository.

`registry` is somehow `null` in here despite of `@NonNull` annotation suggested by Glide: https://github.com/DylanVann/react-native-fast-image/blob/master/android/src/main/java/com/dylanvann/fastimage/FastImageOkHttpProgressGlideModule.java#L51

I wonder if a null check here would prevent those crashes? I can't try it on my own as I don't have a Nexus device. Could you try and check it if you have a Nexus @wsdt?

**Comment by wsdt** [comment]
Unfortunately, I don't have one either. It's just Firebase Crashlytics which is notifying about that issue.

**Comment by systemride** [comment]
I got the same Firebase Crashlytics log. The device mentioned was a Nexus 5X - using Android 6.0.1.

**Comment by alexandru1208** [comment]
Same here. Same crash, same device, same android version, just after uploading a new version on play store, before the app was even available on store.  Probably it's one of the emulators on which they are reviewing the app.

**Comment by halilb** [comment]
Our crashes has the identical environment with what @alexandru1208 described. It's happening only on Google's devices/emulators during the app review. They're fortunately not rejecting the app update because of this.

**Comment by wsdt** [comment]
A simple try-catch should resolve this issue, right? 
Error-handling is never wrong. 

Would be amazing as it falsifies our Playstore & Firebase Crashlytics reports. 

Thank you :)

**Comment by pilot4u** [comment]
we have the same issue.
did anyone managed to solve it?

**Comment by marko-ramsak** [comment]
Crashes happening on Nexus 5X 6.0.1
`java.lang.NullPointerException: Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference`

**Comment by fcaride** [comment]
Same here, its always with the Nexus 5X

**Comment by earlquadra8** [comment]
I got the same thing from the Firebase Crashlytics report.
Does anyone try pulling a null check to see if it works? Or any workaround for this?

I think I can't find a Nexus 5X to test on, hope someone happen to have one and test it. I wonder is it only the emulator or it also happen on a real  Nexus 5X.

**Comment by meosieudang** [comment]
anyone fix it?

**Comment by bbturtle** [comment]
> anyone fix it?

Me too. How to fix?

**Comment by meosieudang** [comment]
> > anyone fix it?
> 
> Me too. How to fix?

@bbturtle I got the issue on Firebase Crashlytics log with device Nexus 5X

**Comment by lesmo** [comment]
This issue is being reported by our Crashlytics, ocurring on App startup and strangely only for Nexus 5X so far:

![image](https://user-images.githubusercontent.com/2228196/73584635-5bbe4700-445f-11ea-8625-ad1f560ed78c.png)

So far no other devices have reported this. An interesting thing for me is the "rooted" info, Nexus 5X is reported as rooted... could it be that the underlying OS is "tampered" to the point of making Glide unstable? Our userbase has no rooted devices so far.

<details>
<summary>Stacktrace</summary>

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'd.b.a.k d.b.a.k.b(java.lang.Class, java.lang.Class, com.bumptech.glide.load.c.v)' on a null object reference
       at com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents(FastImageOkHttpProgressGlideModule.java)
       at com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents(GeneratedAppGlideModuleImpl.java)
       at com.bumptech.glide.Glide.getArrayPool(Glide.java)
       at com.bumptech.glide.Glide.getGlideContext(Glide.java)
       at com.bumptech.glide.Glide.checkAndInitializeGlide(Glide.java)
       at com.bumptech.glide.Glide.clearMemory(Glide.java)
       at com.bumptech.glide.Glide.getContext(Glide.java)
       at com.bumptech.glide.Glide.getBitmapPool(Glide.java)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java)
       at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java)
       at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.addRootView(NativeViewHierarchyManager.java)
       at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java)
       at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java)
       at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java)
       at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:856)
       at android.view.Choreographer.doCallbacks(Choreographer.java:670)
       at android.view.Choreographer.doFrame(Choreographer.java:603)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5421)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

</details>

**Comment by pilot4u** [comment]
same here:
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference
       at com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents(FastImageOkHttpProgressGlideModule.java:51)
       at com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents(GeneratedAppGlideModuleImpl.java:37)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java:273)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java:223)
       at com.bumptech.glide.Glide.checkAndInitializeGlide(Glide.java:184)
       at com.bumptech.glide.Glide.get(Glide.java:168)
       at com.bumptech.glide.Glide.getRetriever(Glide.java:689)
       at com.bumptech.glide.Glide.with(Glide.java:716)
       at com.google.android.libraries.places.internal.fw.b(PG:5)
       at com.google.android.libraries.places.internal.fx.a(PG:2)
       at com.google.android.libraries.places.internal.fx.g(PG:20)
       at com.google.android.libraries.places.internal.fx.h(PG:21)
       at com.google.android.libraries.places.internal.fx.i(PG:23)
       at com.google.android.libraries.places.internal.fx.q(PG:39)
       at com.google.android.libraries.places.internal.fx.a(PG:5)
       at com.google.android.libraries.places.api.Places.a(PG:40)
       at com.google.android.libraries.places.api.Places.createClient(PG:27)
       at com.arttitude360.reactnative.rngoogleplaces.RNGooglePlacesModule.<init>(RNGooglePlacesModule.java:86)
       at com.arttitude360.reactnative.rngoogleplaces.RNGooglePlacesPackage.createNativeModules(RNGooglePlacesPackage.java:29)
       at com.facebook.react.ReactPackageHelper.getNativeModuleIterator(ReactPackageHelper.java:41)
       at com.facebook.react.NativeModuleRegistryBuilder.processPackage(NativeModuleRegistryBuilder.java:41)
       at com.facebook.react.ReactInstanceManager.processPackage(ReactInstanceManager.java:1230)
       at com.facebook.react.ReactInstanceManager.processPackages(ReactInstanceManager.java:1201)
       at com.facebook.react.ReactInstanceManager.createReactContext(ReactInstanceManager.java:1134)
       at com.facebook.react.ReactInstanceManager.access$1000(ReactInstanceManager.java:125)
       at com.facebook.react.ReactInstanceManager$5.run(ReactInstanceManager.java:951)
       at java.lang.Thread.run(Thread.java:818)

**Comment by vikjovanov** [comment]
I got same, any fix ?

**Comment by schlaegerz** [comment]
I tried running this in a nexus 5 simulator and it didn't crash, does any one have any idea how to debug/fix this?

**Comment by lesmo** [comment]
> I tried running this in a nexus 5 simulator and it didn't crash, does any one have any idea how to debug/fix this?

Well, first step is to be able to reproduce an error. That's something nobody has been able to do in an observable way (not happening to anyone outside of what I'd presume is Test Lab environment apparently)...

**Comment by yogodoshi** [comment]
Same error.

Device: Huawei P20 Lite
Android version: 9.1.0
React Native version: 0.61.5
React Native Fast Image version: 7.0.2

**Comment by yogodoshi** [comment]
I'm a newbie in this React Native world so forgive me if I'm saying something stupid but I think I found what's causing the issue (at least for me): Protective Play.

I disabled it and my app was successfully installed.

**Comment by hamidfzm** [comment]
https://github.com/DylanVann/react-native-fast-image/issues/588

**Comment by rtroncoso** [comment]
Fixed in `bumptech/glide` version 4.11.0:

https://github.com/bumptech/glide/issues/3857#issuecomment-530885597

**Comment by DylanVann** [comment]
Awesome, thanks for the PR @rtroncoso !

**Comment by hamidfzm** [comment]
@DylanVann @rtroncoso The issue still persists on after upgrading to 8.1.5 version on Nexus 5x devices.
https://sentry.io/share/issue/f0ab7069d2cd4e03b2d3550bea948223/

**Comment by KSergey65** [comment]
for me issue persist to.
v.8.1.5

NullPointerException
Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference

Android
Version:6.0.1
Nexus 5X
Arch:armeabi-v7a

https://sentry.io/share/issue/a0d2a74ac7cf4c1c9b4da0819c6b5964/

**Comment by ramilexe** [comment]
any updates on that?

**Comment by fgeorgsson** [comment]
Getting this too. Also on Nexus 5X running Android 6.0.1.

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference
       at com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents(FastImageOkHttpProgressGlideModule.java:51)
       at com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents(GeneratedAppGlideModuleImpl.java:34)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java:305)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java:242)
       at com.bumptech.glide.Glide.checkAndInitializeGlide(Glide.java:201)
       at com.bumptech.glide.Glide.get(Glide.java:182)
       at com.bumptech.glide.Glide.getRetriever(Glide.java:749)
       at com.bumptech.glide.Glide.with(Glide.java:776)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java:51)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java:33)
       at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:139)
       at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:83)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:270)
       at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:180)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java:987)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:958)
       at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
       at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
       at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:856)
       at android.view.Choreographer.doCallbacks(Choreographer.java:670)
       at android.view.Choreographer.doFrame(Choreographer.java:603)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5421)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by Eldavid** [comment]
I have a same problem just on Nexus 5X with Android 6.0.1. During developing I didn't get any error connecting with this issue. 

https://sentry.io/share/issue/6d59a68061dd4da1b70b496fc5c079b7/

**Comment by sekizlipenguen** [comment]
I'm having this problem.

 6.0.1
 Nexus 5X

"react-native-fast-image": "^8.1.5"

Fatal Exception: java.lang.NullPointerException
Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference
com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents (FastImageOkHttpProgressGlideModule.java:51)
com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents (GeneratedAppGlideModuleImpl.java:34)
com.bumptech.glide.Glide.initializeGlide (Glide.java:305)
com.bumptech.glide.Glide.initializeGlide (Glide.java:242)
com.bumptech.glide.Glide.checkAndInitializeGlide (Glide.java:201)
com.bumptech.glide.Glide.get (Glide.java:182)
com.bumptech.glide.Glide.getRetriever (Glide.java:749)
com.bumptech.glide.Glide.with (Glide.java:776)
com.dylanvann.fastimage.FastImageViewManager.createViewInstance (FastImageViewManager.java:51)
com.dylanvann.fastimage.FastImageViewManager.createViewInstance (FastImageViewManager.java:33)
com.facebook.react.uimanager.ViewManager.createViewInstance (ViewManager.java:139)
com.facebook.react.uimanager.ViewManager.createView (ViewManager.java:82)
com.facebook.react.uimanager.NativeViewHierarchyManager.createView (NativeViewHierarchyManager.java:269)
com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute (UIViewOperationQueue.java:179)
com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations (UIViewOperationQueue.java:972)
com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:943)
com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:28)
com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:174)
com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:84)
android.view.Choreographer$CallbackRecord.run (Choreographer.java:856)
android.view.Choreographer.doCallbacks (Choreographer.java:670)
android.view.Choreographer.doFrame (Choreographer.java:603)
android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:844)
android.os.Handler.handleCallback (Handler.java:739)
android.os.Handler.dispatchMessage (Handler.java:95)
android.os.Looper.loop (Looper.java:148)
android.app.ActivityThread.main (ActivityThread.java:5421)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:726)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:616)

**Comment by LcsGrz** [comment]
Hey, did anyone find a fix for this? :/

**Comment by fgeorgsson** [comment]
This problem is still very real - affecting our users that have a Nexus 5X device.

**Comment by shekharskamble** [comment]
this should not be closed... same issue here as well...

**Comment by lesmo** [comment]
This got closed because apparently Glide version fixed it:

> Fixed in `bumptech/glide` version 4.11.0:
> 
> [bumptech/glide#3857 (comment)](https://github.com/bumptech/glide/issues/3857#issuecomment-530885597)

It seems it still persists, but perhaps it's better to create a new Issue and link it here too so it gets more attention.

**Comment by nzaghini-phorest** [comment]
We also face identical bug for Nexus 5X devices as of today :(

**Comment by tellodaniel** [comment]
Same issue here:
Android Version: 6.0.1
Nexus 5X
"react-native-fast-image": "8.3.2",
"react-native": "0.62.2",

https://sentry.io/share/issue/60d52bfc98da463e8212aee38f8fc456/

**Comment by tellodaniel** [comment]
@DylanVann can we reopen this pls

**Comment by loiclouvet** [comment]
I confirm having crashes on multiple Nexus 5X devices too :
```
Android Version: 6.0.1
Nexus 5X
react-native-fast-image: 8.3.2,
react-native: 0.63.3
```

@tellodaniel Did you or someone else find a solution / trick to avoid crashes on Nexus 5X ?

**Comment by tellodaniel** [comment]
> I confirm having crashes on multiple Nexus 5X devices too :
> 
> ```
> Android Version: 6.0.1
> Nexus 5X
> react-native-fast-image: 8.3.2,
> react-native: 0.63.3
> ```
> 
> @tellodaniel Did you or someone else find a solution / trick to avoid crashes on Nexus 5X ?

Not yet @loiclouvet :(

**Comment by daheeahn** [comment]
I faced same issue.
my log
`Fatal Exception: java.lang.NullPointerException
Attempt to invoke virtual method 'com.bumptech.glide.Registry com.bumptech.glide.Registry.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.model.ModelLoaderFactory)' on a null object reference`
`com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents (FastImageOkHttpProgressGlideModule.java:51)
com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents (GeneratedAppGlideModuleImpl.java:34)
com.bumptech.glide.Glide.initializeGlide (Glide.java:305)
com.bumptech.glide.Glide.initializeGlide (Glide.java:242)
com.bumptech.glide.Glide.checkAndInitializeGlide (Glide.java:201)
com.bumptech.glide.Glide.get (Glide.java:182)
com.bumptech.glide.Glide.getRetriever (Glide.java:749)
com.bumptech.glide.Glide.with (Glide.java:776)
com.dylanvann.fastimage.FastImageViewManager.createViewInstance (FastImageViewManager.java:51)
com.dylanvann.fastimage.FastImageViewManager.createViewInstance (FastImageViewManager.java:33)
com.facebook.react.uimanager.ViewManager.createViewInstance (ViewManager.java:139)
com.facebook.react.uimanager.ViewManager.createView (ViewManager.java:82)
com.facebook.react.uimanager.NativeViewHierarchyManager.createView (NativeViewHierarchyManager.java:269)
com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute (UIViewOperationQueue.java:179)
com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations (UIViewOperationQueue.java:972)
com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:943)
com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:28)
com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:174)
com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:84)`

**Comment by lesmo** [comment]
> I confirm having crashes on multiple Nexus 5X devices too :
> 
> ```
> Android Version: 6.0.1
> Nexus 5X
> react-native-fast-image: 8.3.2,
> react-native: 0.63.3
> ```
> 
> @tellodaniel Did you or someone else find a solution / trick to avoid crashes on Nexus 5X ?

I'm really curious, have you seen this on a physical device? I've only seen it on Crashlytics, so I didn't think much of it.

**Comment by loiclouvet** [comment]
> @loiclouvet I'm really curious, have you seen this on a physical device? I've only seen it on Crashlytics, so I didn't think much of it.

@lesmo You are totally right, I don't own a Nexus 5X and my Nexus 5X simulator does not crash. No user complained about it yet.
Did you try to contact Crashlytics support in order to investigate a potential false positive ?

**Comment by lesmo** [comment]
> > @loiclouvet I'm really curious, have you seen this on a physical device? I've only seen it on Crashlytics, so I didn't think much of it.
> 
> @lesmo You are totally right, I don't own a Nexus 5X and my Nexus 5X simulator does not crash. No user complained about it yet.
> Did you try to contact Crashlytics support in order to investigate a potential false positive ?

I'm no longer working on that project, but the Issue showed in Crashlytics and the Google Play Developer Console during the automated testing for approval. I was never able to see it on a real device (no access to one) or the Firebase Test Lab. I don't think it's a false positive, but a very rare edge case...

**Comment by alirezatk73** [comment]
we still receive this issue on nexus 5 devices we got the error on Sentry. Fast Image version is ```8.3.2```

**Comment by KestasVenslauskas** [comment]
Still happening in 8.3.4 only on nexus 5 devices, I think it's happening only after enabling proguard, but I followed the rules as provided.

**Comment by dcx86** [comment]
Is there any fix for this?

**Comment by chichilatte** [comment]
Still getting this in March 2021 on Nexus 5X devices with Android 6.0.1. 

Our app is using 
* RN 0.63.3 with hermes js engine
* react-native-fast-image 8.3.4
* enableProguardInReleaseBuilds = false

The stacktrace we get in Bugsnag is...
```
Thread 1 - main
        at com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents(FastImageOkHttpProgressGlideModule.java:49)
        at com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents(GeneratedAppGlideModuleImpl.java:34)
        at com.bumptech.glide.Glide.initializeGlide(Glide.java:305)
        at com.bumptech.glide.Glide.initializeGlide(Glide.java:242)
        at com.bumptech.glide.Glide.checkAndInitializeGlide(Glide.java:201)
        at com.bumptech.glide.Glide.get(Glide.java:182)
        at com.bumptech.glide.Glide.getRetriever(Glide.java:749)
        at com.bumptech.glide.Glide.with(Glide.java:776)
        at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java:51)
        at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java:33)
        at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:139)
        at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:83)
        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:261)
        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:186)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java:1112)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1083)
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:856)
        at android.view.Choreographer.doCallbacks(Choreographer.java:670)
        at android.view.Choreographer.doFrame(Choreographer.java:603)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5421)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

The obvious solution is just exclude the Nexus 5X from list of supported devices in the Google Play console. Shame to do it though!

**Comment by creat-or** [comment]
Still happening, please reopen :)

**Comment by sengchunyee** [comment]
Still happen, please reopen
"react-native-fast-image": "8.3.2",
react-native: 0.63.3

**Comment by alexco2** [comment]
I also have cashes in production with react native version 0.64
Proguard is active

Here is the log:

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'com.bumptech.glide.i com.bumptech.glide.i.r(java.lang.Class, java.lang.Class, com.bumptech.glide.load.q.o)' on a null object reference
       at com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents(FastImageOkHttpProgressGlideModule.java)
       at com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents(GeneratedAppGlideModuleImpl.java)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java)
       at com.bumptech.glide.Glide.initializeGlide(Glide.java)
       at com.bumptech.glide.Glide.checkAndInitializeGlide(Glide.java)
       at com.bumptech.glide.Glide.get(Glide.java)
       at com.bumptech.glide.Glide.getRetriever(Glide.java)
       at com.bumptech.glide.Glide.with(Glide.java)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java)
       at com.dylanvann.fastimage.FastImageViewManager.createViewInstance(FastImageViewManager.java)
       at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java)
       at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java)
       at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java)
       at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java)
       at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java)
       at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:856)
       at android.view.Choreographer.doCallbacks(Choreographer.java:670)
       at android.view.Choreographer.doFrame(Choreographer.java:603)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5421)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by SamiChab** [comment]
Also having this issue on 
```
"react-native": "0.63.3",
"react-native-fast-image": "^8.3.2",
```
Device: Nexus 5X

Please reopen

**Comment by chichilatte** [comment]
It's just one device among many thousands, on an old OS version. And it looks like it's just Google review devices, not real world devices. Cut your losses and just exclude the Nexus 5X from your app's supported devices, that's what that list is for. There are better ways to spend development time (imho).

**Comment by alexco2** [comment]
Thanks for the tipp. I was not aware that this problem just appears on one device. For me that is not a problem then.

**Comment by panda6412** [comment]
Same here !

```
"react-native": "0.63.2",
"react-native-fast-image": "8.3.4", 
Device: Nexus 5X. 
```

Even though appears on one device, but it still need to be fixed.
Please reopen, thanks

**Comment by Pierre-CLIND** [comment]
@panda6412 it's looks like to be an emulator, maybe PlayStore submission bot tests?

**Comment by alexco2** [comment]
I'm still in development and none of my testers have a nexus. I sure this is a Google testing device.

**Comment by lesmo** [comment]
> It's just one device among many thousands, on an old OS version. And it looks like it's just Google review devices, not real world devices. Cut your losses and just exclude the Nexus 5X from your app's supported devices, that's what that list is for. There are better ways to spend development time (imho).

I agree with @chichilatte, those devices are either pretty much dead by now. In the project I worked on we were very interested on these _"legacy"_ devices because part of our target audience included _"low income population"_ which, more often than not, would own old devices like that one.

It's a good idea to try and support all devices, but this particular bug is more trouble than it's worth if nobody has seen it in other, newer, devices. I wouldn't suggest adding it to the exclusion list of the Store, but just ignoring it in your favorite crash reporting tool (Crashlytics, Bugsnag, etc.)

**Comment by zholmes1** [comment]
I was able to resolve this issue by ensuring that I never have a FastImage with a null/undefined uri as a source. Actually I believe this is good practice for even normal Image components from react-native as well.

```
const uri: string | undefined = getYourImageUriFromSomewhere()

// bad
return <FastImage source={{uri}} />

// good
if (uri) {
  return <FastImage source={{uri}} />
}
return null
```

**Comment by ShaeerN-GO** [comment]
same issue here ----

**Comment by ramisalem** [comment]
Same issue from crashlytics

**Comment by esthefanielanza** [comment]
I am still seeing the same issue on crashlytics and sentry.

Devices affected: Nexus 5x
Android Version: 6.0.1

```
"react-native": "0.65.1",
"react-native-fast-image": "8.3.4"
```

**Comment by ShaeerN-GO** [comment]
+1

**Comment by Marialuisaclaro** [comment]
Same Issue

**Comment by alexhtcg** [comment]
> Same Issue

Are you using new version?

**Comment by JeffGuKang** [comment]
It is still happening.  

```
"react-native": "0.66.3",
"react-native-fast-image": "8.5.11",
```

```
 Fatal Exception: java.lang.NullPointerException
Attempt to invoke virtual method 'com.bumptech.glide.j com.bumptech.glide.j.replace(java.lang.Class, java.lang.Class, com.bumptech.glide.load.p.o)' on a null object reference
com.dylanvann.fastimage.FastImageOkHttpProgressGlideModule.registerComponents (FastImageOkHttpProgressGlideModule.java:49)
com.bumptech.glide.GeneratedAppGlideModuleImpl.registerComponents (GeneratedAppGlideModuleImpl.java:34)
com.bumptech.glide.Glide.initializeGlide (Glide.java:329)
com.bumptech.glide.Glide.initializeGlide (Glide.java:266)
....
```

**Comment by davydspotnana** [comment]
Still reproduce on the latest version `"react-native-fast-image": "^8.5.11",`

device: `LGE Nexus 5X (Nexus 5X)`
OS: `Marshmallow (v6)(Level 23)`

**Comment by vladyslav922** [comment]
> repository

Same here, same version "react-native-fast-image": "^8.5.11"

**Comment by thanhdevapp** [comment]
same here Device: Nexus 5X.

**Comment by yamila-fraiman** [comment]
Same issue, any idea?

**Comment by aadityapaliwal94** [comment]
Same issue for me in LGE (Nexus 5X) device.

**Comment by jadonhansen** [comment]
Same issue reported on firebase crashlytics with Nexus 5X. 

`"react-native": "0.64.3"`
`"react-native-fast-image": "^8.5.11"`

Yes @chichilatte we can cut our losses but would also be great to receive some support or feedback from some devs looking at this bug.

**Comment by zholmes1** [comment]
> would also be great to receive some support or feedback from some devs looking at this bug

@jadonhansen Good luck with that one 😂

You'll get used to just ignoring these crash reports. I've accepted it at this point. The only thing that still worries me is whether or not it affects my app store ranking, since reliability is something they are known to consider.

But yeah, getting support on this issue from the devs?

Still waiting 💀

**Comment by kickbk** [comment]
Same

**Comment by ansh** [comment]
Just don’t support the Nexus 5X

**Comment by rishabmahesh** [comment]
The same continues to persists 👀

**Comment by matteodanelli** [comment]
I cnfirm that the issue is still here.. version 8.6.3

**Comment by mohamed2m2018** [comment]
Issue still persists in version 8.6.3

**Comment by Shivemmer** [comment]
any update on this issue ??
